### PR TITLE
Android v2 

### PR DESCRIPTION
**Pull request by mevinDhun**

### CODE REVIEW

File: src/android/UploadPayload.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+    public UploadPayload(String jsonSettings) {
+        try {
+            JSONObject settings = new JSONObject(jsonSettings);
+            filePath = settings.getString("filePath");
+            serverUrl = settings.getString("serverUrl");
+            id = settings.getString("id");
+            fileKey = settings.getString("fileKey");
+            if (settings.has("showNotification"))
```
Comment (ombr) [inline src/android/UploadPayload.java]: Do we really need to manage all this here ?

File: tests/tests.js [hunk 2]
```diff
@@ -283,7 +283,7 @@ exports.defineAutoTests = function () {
 
       it('does not return error if eventId is given', function (done) {
         var nativeUploader = FileTransferManager.init()
-        nativeUploader.acknowledgeEvent('x-coredata://123/UploadEvent/p1', done, null)
+        nativeUploader.acknowledgeEvent(window.cordova.platformId === 'android' ? 'x-coredata://123/UploadEvent/p1' : 123, done, null)
```
Comment (ombr) [inline tests/tests.js]: @mevinDhun du coup on arrive pas a faire un test générique pour ça ?

Comment (mevinDhun) [reply tests/tests.js]: C'est deja corrigé: https://github.com/spoonconsulting/cordova-plugin-background-upload/blob/4c99fcb53e837402b5af86ad6939d437d448b0fe/tests/tests.js#L309
Je crois que t'a regarder une ancienne version

File: src/android/PendingUpload.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            uploadId = payload.getString("id");
+            data = payload.toString();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    public static void create(JSONObject payload) {
+        new PendingUpload(payload).save();
```
Comment (azharbeebeejaun) [inline src/android/PendingUpload.java]:
```suggestion
        PendingUpload pendingUpload = new PendingUpload(payload);
        pendingUpload.save();
        return pendingUpload();
```
?

File: src/android/PendingUpload.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    String data;
+
+    public PendingUpload() {}
+
+    public PendingUpload(JSONObject payload) {
+        try {
+            uploadId = payload.getString("id");
+            data = payload.toString();
+        } catch (JSONException e) {
+            Log.d("CordovaBackgroundUpload", "eventLabel:'error reading id during PendingUpload creation'");
```
Comment (ombr) [inline src/android/PendingUpload.java]: Logs are parsed with key=value :smile: 
```suggestion
            Log.d("CordovaBackgroundUpload", "eventLabel='error reading id during PendingUpload creation'");
```

File: README.md [hunk 5]
```diff
@@ -149,6 +144,7 @@ On android Oreo and above, there are strict limits on background services and it
     });
     ```
 - Events need to be acknowledged to be removed. Failure to do so will always broadcasts the list of saved events on `init`.
+-`showNotification` parameter has been removed (A notification will always be shown on android during upload)
```
Comment (azharbeebeejaun) [inline README.md]:
```suggestion
-`showNotification` parameter has been removed (A notification will always be shown on Android during upload)
```

File: src/android/PendingUpload.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    public PendingUpload(JSONObject payload) {
+        try {
+            uploadId = payload.getString("id");
+            data = payload.toString();
+        } catch (JSONException e) {
+            Log.d("CordovaBackgroundUpload", "eventLabel='error reading id during PendingUpload creation'");
+        }
+    }
+
+
```
Comment (azharbeebeejaun) [inline src/android/PendingUpload.java]:
```suggestion
```

File: tests/tests.js [hunk 7]
```diff
@@ -116,196 +112,190 @@ exports.defineAutoTests = function () {
...
       it('does not return error if eventId is given', function (done) {
-        var nativeUploader = FileTransferManager.init()
+        nativeUploader = FileTransferManager.init({}, function (result) {})
         nativeUploader.acknowledgeEvent('x-coredata://123/UploadEvent/p1', done, null)
       })
+
+      it('persist event id until it is acknowledged', function (done) {
+        nativeUploader = FileTransferManager.init({}, function (event1) {
+          if (event1.state === 'UPLOADED') {
+            var eventId = event1.eventId
```
Comment (azharbeebeejaun) [inline tests/tests.js]:
```suggestion
            var eventId = event1.eventId
            expect(event1.id).toBe('unsub')
```

File: tests/tests.js [hunk 8]
```diff
@@ -116,196 +112,190 @@ exports.defineAutoTests = function () {
...
+            nativeUploader.destroy()
+            nativeUploader = FileTransferManager.init({}, function (event2) {
+              expect(event2.id).toBe('unsub')
+              expect(event2.eventId).toBe(eventId)
```
Comment (azharbeebeejaun) [inline tests/tests.js]:
```suggestion
```

File: www/FileTransferManager.js [hunk 9]
```diff
@@ -1,47 +1,46 @@
...
   }
 
   if (!payload.fileKey) {
     payload.fileKey = 'file'
   }
 
-  if (payload.showNotification === null || payload.showNotification === undefined) {
-    payload.showNotification = true
+  if (!payload.notificationTitle) {
+    payload.notificationTitle = 'Uploading files'
```
Comment (azharbeebeejaun) [inline www/FileTransferManager.js]: payload.notificationTitle = payload.notificationTitle || 'Uploading files'

File: README.md [hunk 10]
```diff
@@ -111,11 +108,9 @@ uploader.on('event', function (event) {
 
 To prevent any event loss while transitioning between native code and javascript side, the plugin stores success/failure events on disk. Once you have received the event, you will need to acknowledge it else it will be broadcasted again when the plugin is initialised. Progress events do not have eventId and are not persisted.
 ```javascript
-uploader.on('event', function(event) {
-    if (event.eventId) {
-        uploader.acknowledgeEvent(event.eventId);
-    }
-});
+if (event.eventId) {
+    uploader.acknowledgeEvent(event.eventId);
```
Comment (azharbeebeejaun) [inline README.md]: success and error callbacks ?

**Pull request merged**